### DISCUSSION TITLE
Question about LDPC and IR-HARQ

### DISCUSSION THREAD

**Discussion by quietinside123** [NVlabs/sionna#250] [Q&A]
# I noticed a passage in LDPC5GEncoder() : 
        #shorten the first 2*Z positions and end after n bits
        #(remaining parity bits can be used for IR-HARQ)
        c_short = tf.slice(c_no_filler, [0, 2*self._z], [batch_size, self.n])
        #incremental redundancy could be generated by accessing the last bits

# How to use this library to complete the operation of IR-HARQ?

**Comment by SebastianCa** [comment]
Hi,

HARQ is currently not supported by the Sionna API and, thus, one would need to extend the encoder/decoder. Please see  [Richardson](https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=8316763) for an overview of the 5G LDPC code construction.

The detailed procedure can be found in Section 5.4.2.1 of TS 38.212:
- For given redundancy version `rv_id`, select `k_0` from Table 5.4.2.1-2 (currently `k_0` is hardcoded to 0)
- Run bit selection procedure from 5.4.2.1 for given `k_0`
- This boils down to slicing the encoded bits from a different slice (cf. comment in the source code)
- At the decoder, one would need to re-combine the received HARQ bits before running the BP decoder

I hope that helps.
